

# README for chrbrolab projects and collaborative projects
# YoSon Park

### last modified 05/14/2018

Here I outline larger project directories and files I have generated between 2015 - early 2018 in the chrbrolab project space. In case of the iPSC project, I outline the work directory in the rader_chrb_lab project space. If no specific github or bitbucket repositories are referenced for the project, I also include some scripts, etc. in addition to the directory-specific README that many follow-up studies will find useful. 


Below are some references one may need in addition to pmacs files if involved in any of the listed studies. In many cases, there are public venues where you may find relevant data so you may not need to access these directories on pmacs.



* For GTEx projects, also check the dbGaP study accession phs000424, e.g. [phs000424.v7.p2](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000424.v7.p2), exchange or [GTEx portal](https://gtexportal.org/home/). 

  * paper to read if using GTEx v6/v6p data: [Genetic effects on gene expression across human tissues.](https://www.ncbi.nlm.nih.gov/pubmed/29022597)


* For raw iPSC project data or many other collaborative projects involving the raderlab members, also check raderlab and rader_chrb_lab project spaces. Raw RNA-seq and genotype data of the iPSC project are uploaded to dbGaP as well along with most other NextGen consortium samples, which our iPSC collection was a part of. One may check the public dbGaP profile under the project's full name [The National Heart, Lung, and Blood Institute (NHLBI)-funded Next Generation Genetic Association Studies (NextGen) Consortium: Phenotyping Lipid traits in iPS derived hepatocytes Study (PhLiPS Study)](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs001341.v1.p1).

  * paper to read if using iPSC data: [Large, Diverse Population Cohorts of hiPSCs and Derived Hepatocyte-like Cells Reveal Functional Genetic Variation at Blood Lipid-Associated Loci.](https://www.ncbi.nlm.nih.gov/pubmed/28388432)


* For ASE project data or collaborative projects involving the bee lab members, check bee tigress spaces.


* For Amish project data or other collaborative projects involving the bucanlab members, check bucanlab or chrb_bucan_lab project spaces


* For raw primates project data, visit NCBI Bioproject database. 

  * paper to read if using primates data: [Transposable elements are the primary source of novelty in primate gene regulation.](https://www.ncbi.nlm.nih.gov/pubmed/28855262)

  * All raw sequence data from this study have been submitted to the [NCBI BioProject database](https://www.ncbi.nlm.nih.gov/bioproject/) under accession numbers [PRJNA349047 (RNA-seq)](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA349047) and [PRJNA349046 (ChIP-seq)](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA349046).
  * There is also a [github repo for some of the analyses scripts mtriz and I generated](https://github.com/ypar/cre_evo_primates)


* For treQTL project data, check [git repo](https://github.com/ypar/treqtl) for the latest update.


* For project data involving splicing analyses, check barashlab project space or relevant bitbucket repositories.



If you have any questions / suggestions, etc., contact me or open an issue. I may get back to you.



## Index

<!--ts-->

* [intro to pmacs; read if you are new to the chrbrolab project space or new to linux](https://github.com/ypar/chrbrolab/blob/master/onboarding/welcome2cdb_v0.1.md)

* [analyses reference data directories](https://github.com/ypar/chrbrolab/blob/master/general/README_reference.md)

* [lab-wide software and resource directories](https://github.com/ypar/chrbrolab/blob/master/general/README_bin.md)

* [lab-wide communications and other setup trivia](https://github.com/ypar/chrbrolab/blob/master/general/README_comm.md)

* [gtex project directories](https://github.com/ypar/chrbrolab/blob/master/gtex/README_gtex.md)

* [ipsc project directories](https://github.com/ypar/chrbrolab/blob/master/ipsc/README_ipsc.md)

* [primates project directories](https://github.com/ypar/chrbrolab/blob/master/primates/README_primates.md)


<!--te-->





